src/core/lv2/Lv2Proc.cpp

```diff
@@ -168,13 +168,35 @@ Plugin::Type Lv2Proc::check(const LilvPlugin *plugin,
 
 
 
+class Lv2ProcSuspender
+{
+public:
+	Lv2ProcSuspender(Lv2Proc* proc)
+		: m_proc(proc)
+		, m_wasActive(proc->m_instance)
+	{
+		if (m_wasActive) { m_proc->shutdownPlugin(); }
+	}
+	~Lv2ProcSuspender()
```

Comment:
This should be marked `noexcept(false)` if `initPlugin()` can throw exceptions, because this destructor is implicitly `noexcept`.

Reply:
[Destructors should never fail](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c36-a-destructor-must-not-fail), so I think a termination on exceptions is practically required and `noexcept`  [should be added](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c37-make-destructors-noexcept).

Reply:
Generally speaking that is true. But for `Lv2ProcSuspender`, given its purpose as a scope guard, and looking at how its data members are plain old data, no issues would be caused by an exception thrown in its destructor. Sort of like the [`std::experimental::scope_success`](https://en.cppreference.com/w/cpp/experimental/scope_success/%7Escope_success) scope guard's destructor which uses a conditional `noexcept`.

But after looking at the implementation of `shutdownPlugin()`, it shouldn't be throwing any exceptions, so `Lv2ProcSuspender` is fine as it is.